Feat/messaging api series (#9917)
firebase/firebase-js-sdk: docs-devsite/messaging_.md

|  <b>function(app, ...)</b> |
|  [getMessaging(app)](./messaging_.md#getmessaging_cf608e1) | Retrieves a Firebase Cloud Messaging instance. |
|  <b>function(messaging, ...)</b> |

> |  [deleteToken(messaging)](./messaging_.md#deletetoken_3fae4b1) | Deletes the registration token associated with this [Messaging](./messaging_.messaging.md#messaging_interface) instance and unsubscribes the [Messaging](./messaging_.messaging.md#messaging_interface) instance from the push subscription.<!-- -->If there is no legacy registration token but the client has FID-based registration metadata (from [register()](./messaging_.md#register_795bb8a)<!-- -->), this deletes that registration on the server, clears local metadata, and invokes [onUnregistered()](./messaging_.md#onunregistered_f8a466e) with the removed FID when successful. |

Reviewer: `and invokes [onUnregistered()](./messaging_.md#onunregistered_f8a466e) with the removed FID when successful.`

`deleteToken` won't invoke `onUnregistered`.  Did you intend to add this comment to `unregister`?

|  [onMessage(messaging, nextOrObserver)](./messaging_.md#onmessage_b9887da) | When a push message is received and the user is currently on a page for your origin, the message is passed to the page and an <code>onMessage()</code> event is dispatched with the payload of the push message. |
|  [onRegistered(messaging, nextOrObserver)](./messaging_.md#onregistered_f8a466e) | Subscribes to an event that the app instance is registered with FCM via Firebase Installation ID (FID). Use the FID passed to the callback to upload it to your application server. When you receive an FID (e.g. after calling [register()](./messaging_.md#register_795bb8a)<!-- -->), instruct your backend to remove any legacy token for this instance. |

> |  [onUnregistered(messaging, nextOrObserver)](./messaging_.md#onunregistered_f8a466e) | Subscribes to an event that the app instance is unregistered from FCM (FID no longer active). Use this to notify your backend to remove this FID to prevent 404 errors on send. |

Reviewer 2: ```suggestion
|  [onUnregistered(messaging, nextOrObserver)](./messaging_.md#onunregistered_f8a466e) | Subscribes to an event that the app instance is unregistered from FCM so the FID is no longer active. Use this to notify your backend to remove this FID to prevent 404 errors on send. |
```

Author: done.

Deletes the registration token associated with this [Messaging](./messaging_.messaging.md#messaging_interface) instance and unsubscribes the [Messaging](./messaging_.messaging.md#messaging_interface) instance from the push subscription.

> If there is no legacy registration token but the client has FID-based registration metadata (from [register()](./messaging_.md#register_795bb8a)<!-- -->), this deletes that registration on the server, clears local metadata, and invokes [onUnregistered()](./messaging_.md#onunregistered_f8a466e) with the removed FID when successful.

Reviewer: Same here.

To stop listening for messages execute this returned function.

### onRegistered(messaging, nextOrObserver) {:#onregistered_f8a466e}

> Subscribes to an event that the app instance is registered with FCM via Firebase Installation ID (FID). Use the FID passed to the callback to upload it to your application server. When you receive an FID (e.g. after calling [register()](./messaging_.md#register_795bb8a)<!-- -->), instruct your backend to remove any legacy token for this instance.

Reviewer 2: ```suggestion
Subscribes to an event that the app instance is registered with FCM via Firebase Installation ID (FID). Use the FID passed to the callback to upload it to your application server. When you receive an FID after calling [register()](./messaging_.md#register_795bb8a)<!-- -->, instruct your backend to remove any legacy token for this instance.
```

<b>Signature:</b>

```typescript
export declare function onRegistered(messaging: Messaging, nextOrObserver: NextFn<string> | Observer<string>): Unsubscribe;
```

#### Parameters

|  Parameter | Type | Description |
|  --- | --- | --- |
|  messaging | [Messaging](./messaging_.messaging.md#messaging_interface) | The [Messaging](./messaging_.messaging.md#messaging_interface) instance. |
|  nextOrObserver | [NextFn](./util.md#nextfn)<!-- -->&lt;string&gt; \| [Observer](./util.observer.md#observer_interface)<!-- -->&lt;string&gt; | A function or observer object called when an FID is registered. |

<b>Returns:</b>

[Unsubscribe](./util.md#unsubscribe)

Unsubscribe function to stop listening.

### onUnregistered(messaging, nextOrObserver) {:#onunregistered_f8a466e}

> Subscribes to an event that the app instance is unregistered from FCM (FID no longer active). Use this to notify your backend to remove this FID to prevent 404 errors on send.

Reviewer 2: ```suggestion
Subscribes to an event that the app instance is unregistered from FCM so the FID is no longer active. Use this to notify your backend to remove this FID to prevent 404 errors on send.
```

<b>Signature:</b>

```typescript
export declare function onUnregistered(messaging: Messaging, nextOrObserver: NextFn<string> | Observer<string>): Unsubscribe;
```

#### Parameters

|  Parameter | Type | Description |
|  --- | --- | --- |
|  messaging | [Messaging](./messaging_.messaging.md#messaging_interface) | The [Messaging](./messaging_.messaging.md#messaging_interface) instance. |
|  nextOrObserver | [NextFn](./util.md#nextfn)<!-- -->&lt;string&gt; \| [Observer](./util.observer.md#observer_interface)<!-- -->&lt;string&gt; | A function or observer object called with the unregistered FID. |

<b>Returns:</b>

[Unsubscribe](./util.md#unsubscribe)

Unsubscribe function to stop listening.

### register(messaging, options) {:#register_795bb8a}

> Registers the app instance with FCM using its Firebase Installation ID (FID). The FID is delivered via the [onRegistered()](./messaging_.md#onregistered_f8a466e) callback, not as a return value. Call this to establish an FID-based identity; once [onRegistered()](./messaging_.md#onregistered_f8a466e) provides an FID, instruct your backend to remove any legacy token previously associated with this instance. The backend send API supports FID as a target.

Reviewer 2: ```suggestion
Registers the app instance with FCM using its Firebase Installation ID (FID). To establish an FID-based identity, call [onRegistered()](./messaging_.md#onregistered_f8a466e). The FID is delivered via the [onRegistered()](./messaging_.md#onregistered_f8a466e) callback, not as a return value. Once you receive an FID, instruct your backend to remove any legacy token previously associated with this instance. The backend send API supports FID as a target.
```
